### ISSUE TITLE
Feature Request: Add quantity to items in ticket details

### ISSUE THREAD

**Issue by Dubz** [osTicket/osTicket#3982]
It would be perfect if I could set up a list of items in the admin panel and add it to the ticket details forum, but be able to add the same item in multiple times.
Right now I have to have multiple fields for however many items I want to allow, which can get outrageous. In other words, if I want to declare a quantity of 10 for a specific item, the system has to have 10 fields set up and I have to enter it into each field.

I know I could use a text field and manually enter data, but that opens the door to typos and human error, versus typing and clicking on an item in a list. Plus it reduces time to type the full name if you can just type 3-4 characters then click the one you want.

**Comment by JediKev** [comment]
Post a screenshot of an example form...because I'm not following.

**Comment by ntozier** [comment]
Yeah i don't follow you at all.  a list is a drop down box.  
So you would use a text field for quantity, and a list for item.

**Comment by Dubz** [comment]
I agree with what you are saying as far as using two fields, one for item and one for quantity.

The problem arises when I enable the allow multiple selections option for the list. I may have two (or more) different items, but some of them could be used more than once.

Let's say, for instance, I was going to add replacement parts for a car (just an example). Now let's say I end up replacing tires, a few bolts, and a few spark plugs. I can add in that tires, bolts, and spark plugs were used by having them selected from the list, but I cannot specify I used 2 tires, 6 bolts, and 3 spark plugs.

The alternative would be to use a text field and allow anything to be typed, but if we were to run a report on how many spark plugs were used, it could be mixed because of how it's typed ("spark plug", "spark plugs", "sparkplugs", or another misspelled version). Someone could also include the model/brand, whereas others don't and keep it simple. Now I'd have to run a search for multiple variations for that item rather than knowing they're all the same.

**Comment by ntozier** [comment]
Or Add a Quantity, and a Drop down list for what you think the maximum amount of parts that could be needed. Agents would populate the data.  If its left blank it will not be displayed on the view ticket view.

**Comment by Dubz** [comment]
Right now I've just added two fields, and may bump it up to three. It will get very cluttered very quickly though, because I need another field for the same entry to mark parts as restocked. So in other words, if I do up to three of the same part, I have to have six fields to mark each one reordered. The issue is it will get cluttered and my coworkers won't like it.

**Comment by JediKev** [comment]
Posting a screenshot of what you mean would help alot...js.

**Comment by Dubz** [comment]
This is a sloppy idea, but what it would sort of look like. Adding a dynamic expandable entry field. You'd click a button (or just have an empty field to add to the bottom), then type a quantity next to it.

https://ss.dubzone.co/1507299432-EuUDpkrV.png

Ignore the "s" after the number, that's just how it displayed. I could've probably just edited the HTML on my browser to display better, but you should hopefully get the idea.


Adding to it, but not a big deal, would to add a second field to it as a checkbox so I can mark it as restocked, but then a simple field starts to turn into more of a custom plugin, so I'm not going to care if it's added as well.

**Comment by Dubz** [comment]
As for storing the data, a simple structure change that would allow this would be to change it from id => name to id => {name: '', quantity: 0}

This would be easy to accommodate the change and shouldn't require too many changes that I'm aware of (as far as changing it across the other parts of the site that displays). Showing it then would be as simple as "[x3] Item name", or however the users would want it. This could be considered a dynamic field attached to a list I presume, since I'm sure someone may want to use quantity at some point for their hookup (I can't be the only one).
